rs/tests/message_routing/queues_compatibility_test.rs

```diff
@@ -252,26 +252,26 @@ fn test(env: TestEnv) {
 
     info!(logger, "Mainnet versions: {:?}", mainnet_versions);
 
-    let tests = mainnet_versions
-        .iter()
-        .map(|v| {
+    let tests = mainnet_versions.iter().flat_map(|v| {
+        [
             TestCase::new(
                 TestType::Bidirectional {
                     published_binary: "replicated-state-test".to_string(),
                     mainnet_version: v.clone(),
                 },
                 "ic/rs/replicated_state/replicated_state_test_binary/replicated_state_test_binary",
                 "canister_state::queues::tests::mainnet_compatibility_tests::basic_test",
-            )
-        })
-        .chain([TestCase::new(
-            TestType::Bidirectional {
-                published_binary: "replicated-state-test".to_string(),
-                mainnet_version: "2d220277bda0b29c90c48eac66fd56beeed59c11".to_string(),
-            },
-            "ic/rs/replicated_state/replicated_state_test_binary/replicated_state_test_binary",
-            "canister_state::queues::tests::mainnet_compatibility_tests::input_order_test",
-        )]);
+            ),
+            TestCase::new(
+                TestType::Bidirectional {
+                    published_binary: "replicated-state-test".to_string(),
+                    mainnet_version: v.clone(),
```

Comment:
Merged this into my [alin/MR-523-response-deduplication](https://github.com/dfinity/ic/tree/alin/MR-523-response-deduplication) branch (#1071) and it fails in the same way it failed before #1105, because mainnet `master` still doesn't have that last change.

I would wait with this until #1105 is on mainnet `master`.

Otherwise LGTM, thanks for following up.

Reply:
Ah. Let's not postpone your PR for 2 more weeks :)
But it is a maintenance burden that I didn't foresee with these comatibility tests, it takes a long time to get rid of mistakes in test cases.

Reply:
Yeah, turns out they're not only backward compatibility, but also forward compatibility tests. (o:

In this case because all the tests / test fixtures we wrote so far never bothered with unique `CallbackIds`. But in the future, it may also be the case that we expect canister queues to have certain properties (e.g. some field populated) before we can safely upgrade them. So not only an annoyance.